### PR TITLE
[2607] Add opt-in undergrad route

### PR DESCRIPTION
**Pull request by JonF1**
### Context
https://trello.com/c/nMF8cKf0/2607-add-opt-in-undergrad-route

### Changes proposed in this pull request
Add opt in undergrad route
### Guidance to review
Create a new trainee and select the 'Opt in (undergrad)' radio button. Check that degree section is not visible on `/review-draft` page and `check-details` page.

### CODE REVIEW

**Comment by JonF1** [comment]
[[M] Add Opt in undergrad route](https://trello.com/c/nMF8cKf0/2607-add-opt-in-undergrad-route)

File: config/settings/development.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ features:
     school_direct_tuition_fee: true
     pg_teaching_apprenticeship: true
     provider_led_undergrad: true
+    opt_in_undergrad: true
```
Comment (defong) [inline config/settings/development.yml:19]: Can we add to here please 

https://github.com/DFE-Digital/register-trainee-teacher-data/blob/2586-s-provider-led-ug-add-the-relevant-bursary/config/settings.yml#L48-L60

File: spec/factories/trainees.rb [hunk 2]
```diff
@@ -209,6 +209,10 @@
       study_mode { COURSE_STUDY_MODES[:full_time] }
     end
 
+    trait :opt_in_undergrad do
+      training_route { TRAINING_ROUTE_ENUMS[:opt_in_undergrad] }
+    end
+
```
Comment (defong) [inline spec/factories/trainees.rb:215]: Is that really needed?

See https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#enum-traits

```ruby
FactoryBot.build(:trainee, :opt_in_undergrad)
```

File: config/settings/development.yml (hunk 1 above)
Comment (JonF1) [reply config/settings/development.yml:19]: it's already on master

File: spec/factories/trainees.rb (hunk 2 above)
Comment (JonF1) [reply spec/factories/trainees.rb]: fixed, thanks

File: config/initializers/training_routes.rb [hunk 3]
```diff
@@ -47,7 +47,8 @@
 }.freeze
 
 TRAINING_ROUTES_FOR_TRAINEE = TRAINING_ROUTES.select { |training_route|
-  TRAINING_ROUTE_ENUMS.values_at(:opt_in_undergrad).exclude? training_route
+  TRAINING_ROUTE_ENUMS.values_at(:assessment_only, :provider_led_postgrad, :early_years_undergrad, :school_direct_tuition_fee, :school_direct_salaried, :early_years_assessment_only,
+                                 :early_years_salaried, :early_years_postgrad, :pg_teaching_apprenticeship, :hpitt_postgrad, :opt_in_undergrad, :provider_led_undergrad).include? training_route
```
Comment (AbigailMcP) [inline config/initializers/training_routes.rb:51]: Are we including everything now? i.e. does `TRAINING_ROUTES_FOR_TRAINEE == TRAINING_ROUTES`

Do we need `TRAINING_ROUTES_FOR_TRAINEE` anymore?

Comment (JonF1) [reply config/initializers/training_routes.rb]: I think so? I've removed in separate commit

**Review by jhenderson — APPROVED** [review]
Looks good.

**Comment by bbelward** [comment]
no degrees in sight, nice work!

**Pull request merged**